canvasapi/discussion_topic.py

```diff
@@ -16,6 +16,9 @@ def _parent_id(self):
         """
         if hasattr(self, "course_id"):
             return self.course_id
+        elif hasattr(self, "context_code"):
+            course_id = self.context_code.split("_")[1]
+            return course_id
```

Comment:
I believe it's possible for the context code to have values other than `course_x`, such as `group_x`.

I think this should also come after the `group_id` check, as the last check before the `else`.

Also, please add test coverage for this section.

Reply:
I went through the docs for announcements, groups, and discussion topics and only announcements had the `context_code` key and it specifically states that only courses are supported right now. I don't think it's a bad idea to add a check...are you okay with a simple `or` in the check?

Reply:
Fair enough!

So long as you're sure there's no way for it to have a `group_id` and `context_code`, then the `or` is fine.